### ISSUE TITLE
Unit conversion test error: conversion from 'nlohmann::json' to non-scalar type 'std::string_view' requested

### ISSUE THREAD

**Issue by taylorhoward92** [nlohmann/json#1634]
- What is the issue you have?

When building v3.6.0 or v3.6.1 I get the following error from a unit-conversion test. The error is not present when building v3.5.0.

```bash
[ 46%] Building CXX object _deps/nlohmann_json-build/test/CMakeFiles/test-conversions.dir/src/unit-conversions.cpp.o
/tmp/tmp.KVk3RBpMpy/cmake-build-debug/_deps/nlohmann_json-src/test/src/unit-conversions.cpp: In function 'void ____C_A_T_C_H____T_E_S_T____0()':
/tmp/tmp.KVk3RBpMpy/cmake-build-debug/_deps/nlohmann_json-src/test/src/unit-conversions.cpp:483:34: error: conversion from 'nlohmann::json' {aka 'nlohmann::basic_json<>'} to non-scalar type 'std::string_view' {aka 'std::basic_string_view<char>'} requested
             std::string_view s = j;
```

- Please describe the steps to reproduce the issue. Can you provide a small but working code example?

Include nlohmann_json in a Cmake project using `FetchContent`:
```
FetchContent_Declare(
        nlohmann_json
        GIT_REPOSITORY https://github.com/nlohmann/json.git
        GIT_TAG v3.6.1
)

FetchContent_MakeAvailable(nlohmann_json)

include_directories(${nlohmann_json_SOURCE_DIR}/single_include)
```

- What is the expected behavior?

Tests pass without errors.

- And what is the actual behavior instead?

The above error occurs.

- Which compiler and operating system are you using? Is it a [supported compiler](https://github.com/nlohmann/json#supported-compilers)?

Ubuntu 18.04 with GCC 8.3.0.
Using C++17.

- Did you use a released version of the library or the version from the `develop` branch?

The test error occurs with v3.6.0 and v3.6.1. It does not occur with v3.5.0. I haven't tested any other versions.

- If you experience a compilation error: can you [compile and run the unit tests](https://github.com/nlohmann/json#execute-unit-tests)?

See above.

**Comment by nickaein** [comment]
Thanks for reporting the issue. I have reproduced here (for 3.6.1 on gcc 8.3): https://godbolt.org/z/pl6XlF

This seems to be an oversight in fixing this unit-test on 3.6.1 which uses implicit conversion and [is discouraged](https://github.com/nlohmann/json/issues/958). It [seems to be fixed](https://github.com/nlohmann/json/blob/4fc98e0b34fb152497f0cd160e58c2f943b19b1e/test/src/unit-conversions.cpp#L465) on develop branch by replacing it with an explicit conversion.

Nevertheless, you should be able to use the library without building unit tests, if you want. Even more, I you can skip downloading the whole repository by replace the git URL with the following direct HTTP link to single header release 3.6.1:

https://github.com/nlohmann/json/releases/download/v3.6.1/json.hpp

The single library header is usually enough for using this library as a third-party in other applications.

**Comment by taylorhoward92** [comment]
Thanks for taking a look. It is fixed on [the line you pointed out](https://github.com/nlohmann/json/blob/4fc98e0b34fb152497f0cd160e58c2f943b19b1e/test/src/unit-conversions.cpp#L465) but not [on this line](https://github.com/nlohmann/json/blob/4fc98e0b34fb152497f0cd160e58c2f943b19b1e/test/src/unit-conversions.cpp#L616).

**Comment by nlohmann** [comment]
Would you like to make a pull request with the fix?

**Comment by nlohmann** [comment]
Closed by merging #1639.
